sdk/cosmosdb/cosmos/src/auth.ts

```diff
@@ -62,6 +62,17 @@ export async function setAuthorizationHeader(
     headers[Constants.HttpHeaders.Authorization] = encodeURIComponent(
       await clientOptions.tokenProvider({ verb, path, resourceId, resourceType, headers })
     );
+  } else if (clientOptions.aadCredentials) {
```

Comment:
Please correct me if I misunderstood.

AFAIU current control flow is, if aadCredentials is set by user then ClientContext created a new bearerTokenAuthenticationPolicy and add its it to the pipeline. 

AADCredental flow is already different from other auth flows. 

Now with this new addition both flows are adding AAD auth header (doing double work). Is this expected/by-design?

Reply:
this shohld have been reverted with the additon of the bearerTokenAuthenticationPolicy in the clinet context.

Reply:
Outdated